packages/components/psammead-social-embed/src/Canonical/index.jsx

```diff
@@ -59,6 +59,20 @@ export const providers = {
 const CanonicalEmbed = ({ provider, oEmbed }) => {
   useScript(providers[provider].script);
 
+  if (provider === 'twitter') {
```

Comment:
- We should try and avoid using hooks inside conditionals - this article is helpful in understanding why: https://reactjs.org/docs/hooks-rules.html.
- One useEffect with this logic inside should be sufficient.
- Is it possible to call `load` or `process` on the relevant embed instead of globally every time one is rendered?
- Instead of hard-coding provider specific logic here inside this component, we could make use of the scripts object and store a callback function as a property that gets called. Similar to the pattern used by `useScript`.

Reply:
I'll look into some of these improvements now, thanks @simonsinclair

Reply:
Hopefully addressed the above here: https://github.com/bbc/psammead/pull/4040/commits/60e835fe25faf247674637a18271f7d6490ea270

> Is it possible to call load or process on the relevant embed instead of globally every time one is rendered?
If I understand your question, from testing I think call the api on the window only enriches the embed in view and not any others that have yet to come into view. I don't know if it 're-enriches' those you have already seen? There is nothing in the docs here about enriching in a more granular way: https://developers.facebook.com/docs/instagram/oembed/